iDT1278 sbml model is not valid and matlab model has no annotations information will add annotation from most recent ecoli model. 

Based and modified from https://github.com/ChristianLieven/Yarrowia_lipolytica_W29-GEM/


In [77]:
from cobra.io import read_sbml_model, load_matlab_model, write_sbml_model

In [78]:
Av_model = load_matlab_model("./iDT1278.mat")
Ec_model = read_sbml_model("./iAF1260.xml")
Kp_model = read_sbml_model("./iYL1228.xml")

cobra\core\reaction.py:446 UserWarning: malformed gene_reaction_rule '((Avin_11760) and (Avin_13240) and (Avin_02950)) or ((Avin_45310) or (Avin_42560)' for <Reaction MPTG at 0x145044f9dc8>
cobra\core\reaction.py:446 UserWarning: malformed gene_reaction_rule '((Avin_11760) and (Avin_13240) and (Avin_02950)) or ((Avin_45310) or (Avin_42560)' for <Reaction MPTG2 at 0x145044f7a88>


In [79]:
#Determine how many metabolites have seed annotation can choose any annotation 

rxn_count = 0
for met in Av_model.metabolites:
    if "seed.compound" not in met.annotation:
        rxn_count +=1
        
print(rxn_count, len(Av_model.metabolites))
        

2003 2003


There are no annotations at all for any metabolite in the model

In [80]:
#Compare annotation of Av model to Ec model
Av_model.metabolites.ac_c.annotation

{}

In [81]:
Ec_model.metabolites.ac_c.annotation

{'sbo': 'SBO:0000247',
 'bigg.metabolite': 'ac',
 'biocyc': 'META:ACET',
 'chebi': ['CHEBI:13704',
  'CHEBI:22165',
  'CHEBI:22169',
  'CHEBI:40480',
  'CHEBI:15366',
  'CHEBI:2387',
  'CHEBI:30089',
  'CHEBI:40486'],
 'envipath': ['5882df9c-dae1-4d80-a40e-db4724271456/compound/9e26dcbe-4db9-46a0-8614-9f03545032d2',
  '5882df9c-dae1-4d80-a40e-db4724271456/compound/5e4989fc-13d3-45d4-ad57-3be380a9d3c0',
  '650babc9-9d68-4b73-9332-11972ca26f7b/compound/a8f0be58-24e8-441b-8d81-a516a0ead4b3',
  '4fd7f3e0-dd25-43ac-9453-dda3e52396e4/compound/d45256fe-61fa-4f5b-bb16-91a3d615e3d8',
  '32de3cf4-e3e6-4168-956e-32fa5ddb0ce1/compound/b545cabc-8c9e-4b20-8848-efa015b481ea',
  '650babc9-9d68-4b73-9332-11972ca26f7b/compound/3e2d750f-df31-4445-9255-163c627e9b4a'],
 'hmdb': 'HMDB00042',
 'inchi_key': 'QTBSBXVTEAMEQO-UHFFFAOYSA-M',
 'kegg.compound': 'C00033',
 'kegg.drug': 'D00010',
 'lipidmaps': 'LMFA01010002',
 'metanetx.chemical': 'MNXM26',
 'reactome.compound': ['1524044', '390305', '113539', '20228

_________________
Match metabolite from ecoli model and A. vinelandii model from id and compartment and rewrite Av annotation with full annotation from Ec

In [82]:
for met in Av_model.metabolites:
    for met_ec in Ec_model.metabolites:
        if met.id in met_ec.id and met.compartment == met_ec.compartment:
            met.annotation.update(met_ec.annotation)

In [83]:
#check for new annotation in Av model 
Av_model.metabolites.ac_c.annotation

{'sbo': 'SBO:0000247',
 'bigg.metabolite': 'sulfac',
 'biocyc': 'META:CPD-10246',
 'chebi': ['CHEBI:34987', 'CHEBI:49876', 'CHEBI:58824', 'CHEBI:50519'],
 'envipath': ['650babc9-9d68-4b73-9332-11972ca26f7b/compound/ed8ac75f-3a91-484c-9d21-d318ef0efef6',
  '650babc9-9d68-4b73-9332-11972ca26f7b/compound/ca85f0e1-e6b7-458b-8f88-d4ed6b580217'],
 'hmdb': 'HMDB01488',
 'inchi_key': 'AGGIJOLULBJGTQ-UHFFFAOYSA-L',
 'kegg.compound': 'C14179',
 'kegg.drug': 'D00049',
 'lipidmaps': 'LMFA01060003',
 'metanetx.chemical': 'MNXM1949',
 'reactome.compound': ['8869604', '197230'],
 'sabiork': '1994',
 'seed.compound': 'cpd09878',
 'slm': '000000449'}

In [84]:
#Test again for number of annotations
rxn_count = 0
for met in Av_model.metabolites:
    if "seed.compound" not in met.annotation:
        rxn_count +=1
        
print(rxn_count, len(Av_model.metabolites))


541 2003


Still 541 metabolites that are not annotatated, These metabolites could be from other models the iDT1278 model was built on. 

\

In [85]:
#run same procedure for reactions
rxn_count = 0
for rxn in Av_model.reactions:
    if "seed.reaction" not in rxn.annotation:
        rxn_count +=1
        
print(rxn_count, len(Av_model.reactions))


2469 2469


In [86]:
#if reaction ids match than write annotation
for rxn in Av_model.reactions:
    for rxn_ec in Ec_model.reactions:
        if rxn.id == rxn_ec.id:
            rxn.annotation.update(rxn_ec.annotation)

In [87]:
#check citrate synthase 
Av_model.reactions.CS.annotation

{'sbo': 'SBO:0000176',
 'bigg.reaction': 'CS',
 'biocyc': ['META:CITSYN-RXN', 'META:RXN-14905'],
 'ec-code': ['2.3.3.16', '2.3.3.1', '2.3.3.3'],
 'kegg.reaction': 'R00351',
 'metanetx.reaction': 'MNXR96920',
 'reactome.reaction': ['R-GGA-373006',
  'R-GGA-70975',
  'R-CEL-70975',
  'R-HSA-70975',
  'R-DRE-70975',
  'R-MMU-70975',
  'R-ATH-70975',
  'R-SPO-70975',
  'R-OSA-70975',
  'R-SCE-70975',
  'R-SSC-70975',
  'R-DDI-70975',
  'R-RNO-70975',
  'R-DME-70975',
  'R-PFA-70975',
  'R-XTR-70975',
  'R-CFA-70975',
  'R-BTA-70975'],
 'rhea': ['16847', '16846', '16845', '16848'],
 'sabiork': '267',
 'seed.reaction': 'rxn00256'}

In [88]:
#recheck reaction annotation count
rxn_count = 0
for rxn in Av_model.reactions:
    if "seed.reaction" not in rxn.annotation:
        rxn_count +=1
        
print(rxn_count, len(Av_model.reactions))

969 2469


969 reactions lack annotaiton 

In [89]:
#Fix malformed MPTG gene
MPTG = Av_model.reactions.get_by_id('MPTG')
MPTG.gene_reaction_rule = '(Avin_11760) and (Avin_13240)'

MPTG2 = Av_model.reactions.get_by_id('MPTG2')
MPTG2.gene_reaction_rule = '(Avin_45310) or (Avin_42560)'

In [90]:
write_sbml_model(Av_model, "./Models/Av_annotate.xml")